File: 153. Find Minimum in Rotated Sorted Array/python/solution.py
class Solution(object):
    def binSearch(self, nums, left, right):
        if nums[left] <= nums[right]:
            return nums[left]
        mid = (left + right) / 2
        if nums[mid] <= nums[right]:
            return self.binSearch(nums, left, mid)
        return self.binSearch(nums, mid + 1, right)
    
    def findMin(self, nums):
        return self.binSearch(nums, 0, len(nums) - 1)

File: 153. Find Minimum in Rotated Sorted Array/java/Solution.java
class Solution {
    public int findMin(int[] nums) {
        return nums[binSearch(nums, 0, nums.length - 1)];
    }
    
    private int binSearch(int[] nums, int i, int j){
        if(nums[i] <= nums[j]) return i;
        
        int m = (i + j) / 2;
        
        if(nums[m] < nums[j]) return binSearch(nums, i, m);;
        
        return binSearch(nums, m + 1, j);
    }
}